client/src/pages/MapPage.jsx

```diff
@@ -1,10 +1,33 @@
-import React from 'react';
+import React, { useState } from 'react';
+import stations from '../data/db.js';
 import Header from '../components/Header.jsx';
+import Map from '../components/Map.jsx';
+import { useMapEvents, Marker, Popup } from 'react-leaflet';
 
 function MapPage({ gasInfoHead, title, fuelValue }) {
+  function LocationMarker() {
+    const [position, setPosition] = useState(null);
+
+    const map = useMapEvents({
+      click() {
+        map.locate();
+      },
+      locationfound(e) {
+        setPosition(e.latlng);
+        map.flyTo(e.latlng, map.getZoom());
+      },
+    });
+    return position === null ? null : (
```

Comment:
I guess this works, but maybe there is an easyer way ? 
`position? do smth : null; `